File: src/ex_21_OOPs_Encapsulation/Lab190_REAL_BANK.java
package ex_21_OOPs_Encapsulation;

public class Lab190_REAL_BANK {
    public static void main(String[] args) {
        ICICIBank amit = new ICICIBank("Amit",100);
        long bal = amit.getBal();
//        System.out.println(amit.bal); can't possible private
        System.out.println(bal);

        // Cashier
        ICICIBank cashier = new ICICIBank("Cash",100);
        cashier.setBal(200,true);
        System.out.println(cashier.getBal());


    }
}

class ICICIBank{
    private String name;
    private long bal;

    public ICICIBank(String name, long bal) {
        this.name = name;
        this.bal = bal;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public long getBal() {
        return bal;
    }

    public void setBal(long bal, boolean isCashier) {
        if(isCashier){
            this.bal = bal;
        }else{
            System.out.println("Not Allowed!!");
        }

    }
}

File: javascript/ex_21_OOPs_Encapsulation/Lab190_REAL_BANK.js
// NOTE: JavaScript uses #prefix for private fields (ES2022). No protected/default access in JS.

class ICICIBank {
    #name;
    #bal;

    constructor(name, bal) {
        this.#name = name;
        this.#bal = bal;
    }

    getName() {
        return this.#name;
    }

    setName(name) {
        this.#name = name;
    }

    getBal() {
        return this.#bal;
    }

    setBal(bal, isCashier) {
        if (isCashier) {
            this.#bal = bal;
        } else {
            console.log("Not Allowed!!");
        }
    }
}

const amit = new ICICIBank("Amit", 100);
const bal = amit.getBal();
// console.log(amit.bal); // can't possible - private
console.log(bal);

// Cashier
const cashier = new ICICIBank("Cash", 100);
cashier.setBal(200, true);
console.log(cashier.getBal());
